src/benchq/resource_estimation/v2/estimators.py

```diff
@@ -0,0 +1,210 @@
+    connected_graph = nx.convert_node_labels_to_integers(connected_graph)
+    scheduler_only_compiler = TwoRowSubstrateScheduler(
+        connected_graph, stabilizer_scheduler=greedy_stabilizer_measurement_scheduler
+    )
+    scheduler_only_compiler.run()
+    return scheduler_only_compiler
+
+
+@dataclass
+class ResourceInfo:
+    synthesis_multiplier: float
+    code_distance: int
+    logical_error_rate: float
+    max_graph_degree: int
+    n_nodes: int
+    n_measurement_steps: int
+    total_time: float
+
+    @property
+    def n_physical_qubits(self) -> int:
+        return 12 * self.max_graph_degree * 2 * self.code_distance**2
+
+    @property
+    def logical_st_volume(self) -> float:
+        return 12 * self.n_nodes * 240 * self.n_nodes * self.synthesis_multiplier
+
+
+class GraphResourceEstimator:
+    def __init__(
+        self, hw_model: BasicArchitectureModel, combine_partition: bool = True
+    ):
+        self.hw_model = hw_model
+        self.combine_partition = combine_partition
+
+    def _logical_operation_error_rate(self, distance: int) -> float:
+        return (
+            distance
+            * 0.3
+            * (70 * self.hw_model.physical_gate_error_rate) ** ((distance + 1) / 2)
+        )
+
+    def _ec_error_rate_synthesized(self, distance: int, n_nodes: int) -> float:
+        return (
+            self._logical_operation_error_rate(distance) * 12 * n_nodes * 240 * n_nodes
+        )
+
+    def _ec_error_rate_unsynthesized(self, distance: int, n_nodes: int) -> float:
+        _, ec_error_rate = balance_logical_error_rate_and_synthesis_accuracy(
+            n_nodes, distance, self.hw_model.physical_gate_error_rate
+        )
+        return ec_error_rate
+
+    def _minimize_code_distance(
+        self,
+        n_nodes: int,
+        error_budget,
+        error_rate: Callable[[int, int], float],
+        min_d: int = 4,
+        max_d: int = 100,
+    ) -> int:
+        target_error_rate = error_budget["total_error"] * error_budget["ec_error_rate"]
+
+        for distance in range(min_d, max_d):
+            if error_rate(distance, n_nodes) < target_error_rate:
+                return distance
+
+        raise RuntimeError(f"Not found good error rates under distance code: {max_d}.")
+
+    def _get_n_measurement_steps(self, graph) -> int:
+        return len(substrate_scheduler(graph).measurement_steps)
+
+    def _estimate_resource_for_graph(
+        self, graph: nx.Graph, n_nodes: int, synthesized: bool, error_budget
+    ) -> ResourceInfo:
+        ec_error_rate = (
+            self._ec_error_rate_synthesized
+            if synthesized
+            else self._ec_error_rate_unsynthesized
+        )
+
+        # Change to code distance
+        code_distance = self._minimize_code_distance(
+            n_nodes, error_budget, ec_error_rate
+        )
+
+        max_degree = max(deg for _, deg in graph.degree())
+        logical_operation_error_rate = (
```

Comment:
nitpick: Why not call `_logical_operation_error_rate`?

Reply:
Cause it's just a variable name?